compiler/crates/relay-lsp/src/goto_definition/mod.rs

```diff
@@ -120,22 +124,48 @@ fn locate_type_definition(
     extra_data_provider: &dyn LSPExtraDataProvider,
     project_name: StringKey,
     type_name: StringKey,
+    schema: &Arc<SDLSchema>,
+    root_dir: &std::path::Path,
 ) -> Result<GotoDefinitionResponse, LSPRuntimeError> {
+    let type_ = schema.get_type(type_name);
+
+    let schema_location = type_.and_then(|type_| {
+        match type_ {
+            // Why don't these other types have locations?
```

Comment:
@captbaritone / @alunyov can you speak to this?

Reply:
I think we just haven't added it yet, but we should! Will require adding the location field to the correct struct within the schema, and then collecting it during parse.

Reply:
okay that makes sense. this sounds pretty deep to fix maybe? you think I could take a shot at this?

Reply:
Yeah. It can be a pretty tedious update, since you need to touch every place that creates or updates the struct field that you choose to update to be `WithLocation`, but it tends to be pretty mechanical if you have Rust Analyzer setup correctly.

1. Update the field to be `WithLocation<_>`
2. Fix all errors by either accessing `.item` off the field, or using some version of `WithLocation::new` etc.

Once cargo check returns cleanly, you are done (I'll have some other work internally to update our other crates, but i"m happy to do that).

Want to take a stab at one?

Reply:
yup! I'll take a stab and let you know if anything gets in the way